wire-ios-sync-engine/Source/E2EI/CRL/ProxyCRLProvider.swift

```diff
@@ -0,0 +1,54 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// Depending on the `mlsE2EId` feature flag configuration, we should use a CRL proxy or fetching the CRL directly.
+/// https://wearezeta.atlassian.net/wiki/spaces/PAD/pages/1147666542/2024-04-22+CRL+proxy+for+mobile+apps
+struct ProxyCRLProvider {
```

Comment:
From the outside, it  seems like this object will always provide a proxy URL because of the name `ProxyCRLProvider`, but actually it may or may not depending on private flag.

I would suggest one of two things:
1. Rename the struct to `CRLURLProvider` or something like that
2. Build the crl url in a private method in the `CertificateRevocationListsChecker`, since it appear that is the only place that needs it, and delete this struct.